main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"log"
+	"version-action/pkgs/version"
+	"version-action/pkgs/utils"
+)
+
+var commitTypes = [...]string {"breaking", "feature", "bugfix"}
+var bumps = map[string]string{"breaking": "major", "feature": "minor", "bugfix": "patch"}
+
+type Commit struct {
+	Tag string
+	Commit string
+}
+
+func prepareTagCommit(commitMessage, environment string) Commit {
+	commit := Commit{}
+	// Get latest tag
+	latestTagRaw, err := version.GetLatestTag(environment)
+	if err != nil {
+		log.Fatalln("Error was found while getting the latest commit message", err)
+	}
+	// Trim tag & Remove 'v'
+	commit.Tag = version.TrimTag(latestTagRaw)
+
+	// Get commit message version level (breaking, feature, bugfix)
+	versionType := version.GetVersionType(commitMessage, commitTypes)
+	if !versionType.Contains {
+		log.Fatalln("Version tag is not valid. Commit message must contain one of the following: [breaking, feature, bugfix]")
+	}
+	commit.Commit = versionType.VersionType
+	return commit
+}
+
+func stagingVersion(tag, verionType string, rc bool) string {
+	log.Println("Building staging version...")
+	// increase rc version by 1
+	if rc {
+		rcTag, err := version.IncreaseRc(tag)
+		if err != nil {
+			log.Fatalln("Error converting rc version to int:", err)
+		}
+		finalTag := version.AddV(rcTag)
+		return finalTag
+	// Bump version and add rc
+	} else {
```

Comment:
why do you need if else here ? 
inside the if you are returning else is redundant 
please take the code out of the else statement